#165273572 :  User Should retrieve and edit their profile

nadralia opened the pull request:
#### What does this PR do?

- Create a user's profile on successful user registration
- Enable a user view their profile
- Enable a user edit their profile
- Allow a user view other users profiles
- Prevent a user from editing other users profiles

#### How should this be manually tested?

- Clone the repo onto you're local machine or fetch the branch
    `git clone https://github.com/andela/ah-backend-dojo.git`
- Move into the project directory
    `cd ah-backend-dojo`
- Create and activate virtual environment
   `virtualenv yourvirtualname`
   `source yourvitualname/bin/activate`
- Install the dependencies   
 `pip3 install -r requirements.txt `
- Run the application
 `python3 manage.py runserver`
- Open postman
- Register a user using the route
    `http://127.0.0.1:8000/api/users/`
- Login the user you just registered
   `http://127.0.0.1:8000/api/users/login/`
- Copy the token that was returned
- Insert a header in postman with a key of Authorization and value Bearer <token> and paste your token
- Then visit the route http://127.0.0.1:8000/api/profiles/username/ to view user profile using GET.
- Then visit the route http://127.0.0.1:8000/api/profiles/username/ to edit the user profile using PUT.



#### What are the relevant pivotal tracker stories?
[#165273572](https://www.pivotaltracker.com/story/show/165273572)

phillipseryazi commented:
@nadralia you have four commits on your pr

nadralia commented:
> @nadralia you have four commits on your pr

Am squashing it

bisonlou commented on authors/apps/profiles/models.py:
```diff
@@ -0,0 +1,38 @@
+    lastname = models.CharField(max_length=250, default='')
+    # display some info about a user
+    bio = models.TextField(blank=True)
+    username = models.CharField(max_length=50, unique=True)
```
Having a username in the profile table creates data redundancy as we already have the username in the User table.

nadralia replied:
Done Working On it .

ZackAtama reviewed (changes requested):
Hello @nadralia, Kindly work on the travis build.

kalsmic commented on authors/apps/profiles/views.py:
```diff
@@ -0,0 +1,46 @@
+            profile, data=request.data, partial=True)
+        if serializer.is_valid(raise_exception=True):
+            serializer.save()
+            return Response({"profile": serializer.data}, status=status.HTTP_200_OK)
```
Please update the response to 
```
return Response( serializer.data, status=status.HTTP_200_OK)
```
Instead of 
```
{
    "profile": {
        "username": "jake",
        "bio": "I work at statefarm",
        "image": "image-link",
        "following": false
    }
}
```
it is showing
```
{
    "profile": {
        "profile": {
            "username": "jake",
            "bio": "I work at statefarm",
            "image": "image-link",
            "following": false
        }
    }
}
```

kalsmic replied:
Well done @nadralia, feedback implemented 👍

kalsmic reviewed (changes requested):
> Please update the response to
> 
> ```
> return Response( serializer.data, status=status.HTTP_200_OK)
> ```
> Instead of
> 
> ```
> {
>     "profile": {
>         "username": "jake",
>         "bio": "I work at statefarm",
>         "image": "image-link",
>         "following": false
>     }
> }
> ```
> it is showing
> 
> ```
> {
>     "profile": {
>         "profile": {
>             "username": "jake",
>             "bio": "I work at statefarm",
>             "image": "image-link",
>             "following": false
>         }
>     }
> }
> ```
Please update the response for successful profile update

nadralia commented:
> Please update the response for successful profile update

Done thanks

JamesMudidi reviewed (changes requested):
Adrali, Please work on the build for travis

nadralia committed:
Ft(profile): update and view profile

Add update profile
Add view profile
write tests for view/update

[Finishes #165273572]

nadralia commented:
> Adralia, Please work on the build for travis

Fixed it

nadralia commented:
> Hello @nadralia, Kindly work on the travis build.

Fixed it

JamesMudidi reviewed (approved):
Merge Approved
